.ci/configure_sharp.sh

```diff
@@ -0,0 +1,335 @@
+    echo "FAIL"
+    exit 1
+fi
+
+CONFIGURE_SHARP_TMP_DIR="${NFS_WORKSPACE}/configure_sharp_$$"
+mkdir -p ${CONFIGURE_SHARP_TMP_DIR}
+
+CFG_DIR="${SCRIPT_DIR}/cfg"
+HOSTFILE="${CFG_DIR}/$HOSTNAME/hostfile"
+export SHARP_CONF="${CONFIGURE_SHARP_TMP_DIR}"
+export SHARP_INI_FILE="${SHARP_CONF}/sharp_manager.ini"
+
+cp -R ${CFG_DIR}/$HOSTNAME/sharp_conf/* ${SHARP_CONF}
+
+if [ ! -f "${HOSTFILE}" ]
+then
+    echo "ERROR: ${HOSTFILE} doesn't exist or not accessible"
+    echo "FAIL"
+    exit 1
+fi
+
+if [ -f "${SHARP_CONF}/sharp_am_node.txt" ]
+then
+    SHARP_AM_NODE=`cat ${SHARP_CONF}/sharp_am_node.txt`
+    echo "INFO: SHARP_AM_NODE = ${SHARP_AM_NODE}"
+else
+    echo "ERROR: ${SHARP_CONF}/sharp_am_node.txt does not exist or not accessible"
+    echo "FAIL"
+    exit 1
+fi
+
+IB_DEV="mlx5_0"
+SM_GUID=`sudo sminfo -C ${IB_DEV} -P1 | awk '{print $7}' | cut -d',' -f1`
+# SM/AM node
+# SM_HOSTNAME=`sudo ibnetdiscover -H -C mlx5_0 -P1  | grep ${SM_GUID} | awk -F'"' '{print $2 }' | awk '{print $1}'`
+HOSTS=`cat $HOSTFILE | xargs | tr ' ' ','`
+
+echo "INFO: HOSTFILE = ${HOSTFILE}"
+echo "INFO: IB_DEV = ${IB_DEV}"
+echo "INFO: SM_GUID = ${SM_GUID}"
+# echo "INFO: SM_HOSTNAME = ${SM_HOSTNAME}"
+echo "INFO: HOSTS = ${HOSTS}"
+
+rm -f ${SHARP_INI_FILE}
+
+cat > ${SHARP_INI_FILE} <<EOF
+sharp_AM_server="${SHARP_AM_NODE}"
+sharp_am_log_verbosity="3"
+sharp_hostlist="$HOSTS"
+sharp_manager_general_conf="${SHARP_CONF}"
+sharpd_log_verbosity="3"
+EOF
+
+echo "INFO: SHARP_INI_FILE ${SHARP_INI_FILE} BEGIN"
+cat ${SHARP_INI_FILE}
+echo "INFO: SHARP_INI_FILE ${SHARP_INI_FILE} END"
+
+check_opensm_status() {
+    echo "Checking OpenSM status on ${SHARP_AM_NODE}..."
+
+    ssh "${SHARP_AM_NODE}" "systemctl status opensmd"
+    if [ $? -ne 0 ]
+    then
+        echo "ERROR: opensmd is not run on ${SHARP_AM_NODE}"
+        echo "FAIL"
+        exit 1
+    fi
+
+    echo "Checking OpenSM status on ${SHARP_AM_NODE}... DONE"
+}
+
+check_opensm_conf() {
+    echo "INFO: check_opensm_conf on ${SHARP_AM_NODE}..."
+
+    OPENSM_CONFIG="/etc/opensm/opensm.conf"
+    echo "INFO: opensm config = ${OPENSM_CONFIG}"
+
+    ssh "${SHARP_AM_NODE}" "grep \"routing_engine.*updn\" ${OPENSM_CONFIG} 2>/dev/null"
+    if [ $? -ne 0 ]
+    then
+        echo "ERROR: wrong value of routing_engine parameter in ${OPENSM_CONFIG}"
+        echo "Should be (example): routing_engine updn"
+        echo "FAIL"
+        exit 1
+    fi
+
+    ssh "${SHARP_AM_NODE}" "grep \"sharp_enabled.*2\" ${OPENSM_CONFIG} 2>/dev/null"
+    if [ $? -ne 0 ]
+    then
+        echo "ERROR: wrong value of sharp_enabled parameter in ${OPENSM_CONFIG}"
+        echo "Should be (example): sharp_enabled 2"
+        echo "FAIL"
+        exit 1
+    fi
+
+    echo "INFO: check_opensm_conf on ${SHARP_AM_NODE}... DONE"
+}
+
+verify_sharp() {
+    echo "INFO: verify_sharp..."
+
+    export PATH="${SHARP_DIR}/bin:$PATH"
+    export LD_LIBRARY_PATH="${SHARP_DIR}/lib:${NCCL_DIR}/lib:${LD_LIBRARY_PATH}"
```

Comment:
it should be set by ```module load hpcx-gcc```
maybe nccl vars should be placed into /path/to/nccl/etc/nccl.module and load it?

well done PR!

Reply:
Fixed for SHARP_DIR - now we fully rely on `module load hpcx-gcc` and HPCX_SHARP_DIR.
For NCCL_DIR I'm planning to implement nightly NCCL testing and NCCL env may be set after that via something like `module load dev/nccl-nightly`.